media-sound/denemo/denemo-2.2.0.ebuild

```diff
@@ -0,0 +1,126 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit xdg-utils
+
+DESCRIPTION="A music notation editor"
+HOMEPAGE="http://www.denemo.org/"
+SRC_URI="mirror://gnu/${PN}/${P}.tar.gz"
+
+LICENSE="GPL-3 OFL-1.1"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+# configure options currently not used:
+# --enable-mem(no) memory debugging: needs Electric fence (efence), which
+#		is not available in portage. See https://github.com/boundarydevices/efence
+# --enable-gtk-doc-pdf(no) doesn't work
+IUSE="alsa +aubio coverage debug +evince doc jack +fluidsynth gtk3 nls \
+	+portaudio +portmidi +rubberband static test"
+
+RDEPEND="
+	dev-libs/libxml2:2
+	>=dev-scheme/guile-2:12
+	gnome-base/librsvg:2
+	media-libs/fontconfig:1.0
+	>=media-libs/libsmf-1.3
+	>=media-libs/libsndfile-1.0.28-r1
+	>=media-sound/lilypond-2.19.54[guile2]
+	alsa? ( >=media-libs/alsa-lib-1.1.2 )
+	aubio? ( >=media-libs/aubio-0.4.1-r1 )
+	evince? ( >=app-text/evince-3.22.1-r1 )
+	fluidsynth? ( >=media-sound/fluidsynth-1.1.6-r1 )
+	gtk3? (
+		x11-libs/gtk+:3
+		x11-libs/gtksourceview:3.0
+	)
+	!gtk3? (
+		x11-libs/gtk+:2
+		x11-libs/gtksourceview:2.0
+	)
+	jack? ( virtual/jack )
+	portaudio? (
+		>=media-libs/portaudio-19_pre20140130
+		sci-libs/fftw:3.0
+	)
+	portmidi? ( >=media-libs/portmidi-217-r1 )
+	rubberband? ( >=media-libs/rubberband-1.8.1-r1 )
+"
+
+DEPEND="${RDEPEND}
+	>=dev-util/intltool-0.51.0-r1
+	>=sys-devel/flex-2.6.1
+	virtual/pkgconfig
+	virtual/yacc
+	doc? ( >=dev-util/gtk-doc-1.25-r1 )
+	nls? ( >=sys-devel/gettext-0.19.8.1 )"
+
+DOCS=( AUTHORS ChangeLog docs/{DESIGN{,.lilypond},GOALS,TODO} NEWS )
+#HTML_DOCS=( docs/denemo-manual.html docs/denemo.css )
+
+PATCHES=( "${FILESDIR}/${P}-no-portaudio-fix.patch" )
+
+src_prepare() {
+	sed -ie '/^Categories=/s/GNOME\;/GNOME\;GTK\;/' pixmaps/denemo.desktop
```

Comment:
`|| die`. Also `-ie` doesn't do what you think it does (try it!).

Reply:
Added `|| die`. 
`-ie` seems to do exactly what I want it to do.
```
waebbl@artus /var/tmp/portage/media-sound/denemo-2.2.0/work/denemo-2.2.0 $ grep Categories pixmaps/denemo.desktop 
Categories=GNOME;Audio;AudioVideo;Music;Education;
waebbl@artus /var/tmp/portage/media-sound/denemo-2.2.0/work/denemo-2.2.0 $ sed -ie '/^Categories=/s/GNOME\;/GNOME\;GTK\;/' pixmaps/denemo.desktop 
waebbl@artus /var/tmp/portage/media-sound/denemo-2.2.0/work/denemo-2.2.0 $ grep Categories pixmaps/denemo.desktop 
Categories=GNOME;GTK;Audio;AudioVideo;Music;Education;
```
I also verified it by doing `ebuild denemo-2.2.0 clean prepare` with the same result.

A different question would be, whether GTK is a valid category, or if this category is used by gentoo. If you're referring to this, I'd be glad to remove the line. It wasn't added by me in the first place, but AFAIR was from an ebuild in bug #515888, where I just took it over and verified, that it's technically working.
I haven't used GNOME for years, but on my xfce desktop at least, there's no such category, but then there's also no category GNOME, Audio, AudioVideo and Music. Didn't check with lxqt or enlightenment, though, both of which I use here and then.

Reply:
`-ie`!=`-i -e`

Reply:
Yes, I was thinking you're referring to this, but still `-ie` does the same as '-i -e' at least on my machine, having sed-4.5. Maybe it's arch dependant, so I change it to `-i -e`.

Reply:
`-ie` means to create backup file with 'e' suffix. `-e` is optional in this case, so you don't notice it didn't work.

Reply:
Ouch... Sure, that's also what the man page is telling me. No, I didn't notice, because I was not checking the files system.
Thanks for pointing this out! :+1: